lib/news_scraper/uri_parser.rb

```diff
@@ -0,0 +1,15 @@
+module NewsScraper
+  class URIParser
```

Comment:
We'll need to do some funky check to see if the scheme already exists if we want to use `URI.host` http://stackoverflow.com/a/6674363/4463793.

I also wanted to write my own regex and this is much more succinct in my opinion.

We need to filter the scheme (and then add `http://` https://github.com/richardwu/news_scraper/blob/master/lib/news_scraper/extractors/article.rb#L15) before sticking it into `HTTParty` because it'll try to connect via pure TCP without HTTP, and some urls may not come prepended with `http://`.